core/domain/exp_services.py

```diff
@@ -1578,53 +1650,92 @@ def regenerate_exploration_and_contributors_summaries(exploration_id):
         exploration_id: str. ID of the exploration.
     """
     exploration = exp_fetchers.get_exploration_by_id(exploration_id)
-    exp_summary = _compute_summary_of_exploration(exploration)
-    exp_summary.contributors_summary = (
-        compute_exploration_contributors_summary(exp_summary.id))
-    save_exploration_summary(exp_summary)
-
-
-def _compute_summary_of_exploration(exploration):
-    """Create an ExplorationSummary domain object for a given Exploration
-    domain object and return it.
+    exp_rights = rights_manager.get_exploration_rights(
+        exploration_id, strict=True)
+    exp_summary = exp_fetchers.get_exploration_summary_by_id(
+        exploration_id, strict=True)
+    updated_exp_summary = update_exploration_summary(
+        exploration, exp_rights, exp_summary)
+    updated_exp_summary.contributors_summary = (
+        compute_exploration_contributors_summary(updated_exp_summary.id))
+    save_exploration_summary(updated_exp_summary)
+
+
+def update_exploration_summary(
+    exploration,
+    exp_rights,
+    exp_summary,
+    skip_exploration_model_last_updated=False
+):
+    """Updates an exploration summary domain object from a given exploration
+    and its rights.
 
     Args:
         exploration: Exploration. The exploration whose summary is to be
             computed.
+        exp_rights: ActivityRights. The exploration rights model, used
+            to compute summary.
+        exp_summary: ExplorationSummary. The exploration summary
+            model whose summary needs to be recomputed.
+        skip_exploration_model_last_updated: bool. Whether the update of
+            exploration_model_last_updated should be skipped.
+            The exploration_model_last_updated is computed from the last human
+            update of the exploration. This can be skipped  in cases where we
```

Comment:
... The update for this value should be skipped when we know that the current workflow isn't due to a human-initiated update.

Reply:
Done.